src/main/scala/PacketAssembler/disassembler.scala

```diff
@@ -134,7 +135,7 @@ class PacketDisAssembler extends Module {
 
   //Handshake Parameters
   val out_valid = RegInit(false.B)
-  val out_fire = io.out.ready & io.out.valid
+  val out_fire = io.out.data.ready & io.out.data.valid
   val in_ready = RegInit(Bool(), false.B)
   val in_fire = io.in.ready & io.in.valid
```

Comment:
```suggestion
  val in_fire = io.in.ready && io.in.valid
```

In general if it's a bool I like to use the `&&` operator instead of bitwise and `&`